lib/broadside/configuration.rb

```diff
@@ -57,7 +57,10 @@ def ssh_cmd(ip, options = {})
       cmd << " -i #{@ssh[:keyfile]}" if @ssh[:keyfile]
       if (proxy = @ssh[:proxy])
         raise MissingVariableError, "Bad proxy: #{proxy[:host]}/#{proxy[:port]}" unless proxy[:host] && proxy[:port]
-        cmd << " -o ProxyCommand=\"ssh #{proxy[:host]} nc #{ip} #{proxy[:port]}\""
+        cmd << ' -o ProxyCommand="ssh -q'
+        cmd << " -i #{proxy[:keyfile]} " if proxy[:keyfile]
```

Comment:
wait you won't get that trailing space if there's no keyfile; i think you need a separate `cmd << ' '` line